server/athenian/api/internal/features/github/pull_request_filter.py

```diff
@@ -1366,3 +1368,58 @@ async def noop():
         # we need it inside df_from_structs()
         PullRequestJiraMapper.apply_empty_to_pr_facts(facts)
     return filtered_prs, dfs, facts, matched_bys, deployments_task
+
+
+def pr_facts_stages_masks(pr_facts: pd.DataFrame) -> npt.NDArray[int]:
+    """Given a df of PullRequestFacts columns return the masks representing their stages.
+
+    Each mask is an integer where the bit at position PullRequestEvent.STAGE.value is 1.
+    This function returns the same stages collected by
+    PullRequestListMiner._collect_events_and_stages.
```

Comment:
Can we replace `_collect_events_and_stages` with this function? Or rely on it, at least. Just asking, for the sake of DRY.

Reply:
It's not immediate since `PullRequestListMiner`  is handling PR one by one. 

If `dfs.prs` received by `PullRequestListMiner` matches the PR it iterates (received `prs`) I guess we can calculate the masks with this function before the iteration and then use the pr mask in `_collect_events_and_stages` to get stages for each PR.

Maybe we can create a tech debt issue to refactor this ?

Reply:
All right. Regarding the refactoring, we are going to create a lot of these issues soon, according to the new team strategy. We can probably ignore this for now.